### PR TITLE
Change HD Wallet derivation path to align with BIP32/BIP44 like others in the ecosystem (i.e Ledger)

### PR DESCRIPTION
**Pull request by fproulx-dfuse**
This PR propose to change the HD Wallet derivation path to align with majority of other wallets.

The gist is that we use BIP44 in the same way that `Ledger Live` does (i.e. derivation path in the form `m / purpose' / coin_type' / ACCOUNT' / change / address_index`), where ACCOUNT is a variable that changes based on the dropdown action in the extension.

**Background:**
- `solana-keygen` simply uses bip39 from tiny-bip39  and uses the provided passphrase to salt the BIP39 derivation (PBKDF2 on top) https://github.com/solana-labs/solana/blob/679e7863cb36fffa8bb62d179a37536fb076047f/clap-utils/src/keypair.rs#L227
- solana remote-wallet uses BIP32 when talking with Ledger, etc. with m/44'/501'/<account>'/<change>'  …
https://github.com/solana-labs/solana/blob/b0d1c70718d58509d95e1bee80fb4d4f5b6a2eae/remote-wallet/src/remote_wallet.rs#L209
- This extension code prior to this PR used bip32 with derivation path in the form `m/501'/${this.walletIndex}'/0/${accountIndex}`  and does not use the passphrase in the derivation (only for on-disk wallet encryption)
- spl-token-wallet  uses BIP32 with derivation path in the form `m/501'/${walletIndex}'/0/${accountIndex}`  and does not use the passphrase in the derivation (only for on-disk wallet encryption)
https://github.com/project-serum/spl-token-wallet/blob/9643de889ba3c6968fc2e777e50f778fdf569b57/src/utils/wallet.js#L37

**References:**
*Announcement of the BIP44 change by Ledger Live team*
https://github.com/MyCryptoHQ/MyCrypto/issues/2070#issue-341249164
*Inventory of derivation path for hardware wallets*
https://medium.com/myetherwallet/hd-wallets-and-derivation-paths-explained-865a643c7bf2

### CODE REVIEW

File: app/background/lib/wallet.ts [hunk 1]
```diff
@@ -21,16 +20,19 @@ export class Wallet {
...
   addAccount() {
     const accountIndex = this.accounts.length
     log("Adding account to wallet with index %s", accountIndex)
+		// Ref. We align on BIP44 like Ledger's most recent Live products now do
+		// https://medium.com/myetherwallet/hd-wallets-and-derivation-paths-explained-865a643c7bf2
+		// >> https://github.com/MyCryptoHQ/MyCrypto/issues/2070#issue-341249164
     const derivedSeed = bip32
       .fromSeed(this.seed)
-      .derivePath(`m/501'/${this.walletIndex}'/0/${accountIndex}`).privateKey
+			.derivePath(`m/44'/501'/${accountIndex}'/0/0`)
```
Comment (t-nelson) [inline app/background/lib/wallet.ts]: +1 the move to BIP44.

I think all path components should be of the hardened variety, however.  This is due to quirks with Ed25519 not playing well with BIP32.  That is, I think the path here should be `m/44'/501'/${accountIndex}'/0'/0'`

**Review by t-nelson — COMMENTED** [review]
Looks good over all!

One nit.  I'm not sure what the conventions of this repository are, but it looks like these change introduce leading tabs :scream:

File: app/background/lib/wallet.ts (hunk 1 above)
Comment (CriesofCarrots) [reply app/background/lib/wallet.ts]: Ditto everything @t-nelson said.

Not sure if you are interested in users being able to derive/manage the same accounts using a Ledger device, but the ledger-app-solana only supports 2 levels of derivation beyond 501'. So if you've no need for the bip32 "account_index", perhaps `m/44'/501'/${accountIndex}'/0'`?

Comment (fproulx-dfuse) [reply app/background/lib/wallet.ts]: @t-nelson @CriesofCarrots we are happy to align with your recommendations. 

Yes, as per Anantoly's recommendation we want to align as much with Ledger. So I assume we're aligning with the Ledger Solana app
https://github.com/solana-labs/ledger-app-solana/blob/c66543976aa8171be6ea0c0771b1e9447a857c40/examples/example-sign.js#L57-L83v

We harden all path components and we drop the last part as it's not much relevant in Solana.

`m/44'/501'/${accountIndex}'/0'`

File: src/background/__tests__/wallet-derivation.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		// m / 44'      / 501'       / [VARIABLE]' / 0'      / [ABSENT]
+
+		const assertDerivation = function(account, expectedSeed, expectedAccount) {
+			const bip32Path_0 = bip32.fromSeed(bufSeed).derivePath(`m/44'/501'/${account}'/0'`)
+			expect(bip32Path_0.publicKey.toString("hex")).toEqual(expectedSeed)
+			const naclPubKey = nacl.sign.keyPair.fromSeed(new Uint8Array(bip32Path_0.privateKey)).publicKey
+			expect(bs58.encode(naclPubKey)).toEqual(expectedAccount)
+		}
+
+		assertDerivation(0, "02e562aad7d744da37474d3440c5716d664683e6410a611cea22b7f07983840809", "FrJPDJphtfZzgobT7b2CG9Mp79D6gVHR8L66T9mQ59Hu")
```
Comment (fproulx-dfuse) [inline src/background/__tests__/wallet-derivation.test.ts:45]: @t-nelson @CriesofCarrots do you have some test vectors to validate compatibility with your Ledger app ?

Comment (CriesofCarrots) [reply src/background/__tests__/wallet-derivation.test.ts:45]: I can't think of a great way to do that, short of plugging in a ledger restored from the same mnemonic and comparing public keys.
If you go that route, here's an integration test that could be modified to generate those keys: https://github.com/solana-labs/ledger-app-solana/blob/d207ec2bcea2613619dbf00e15d986d07cf24922/tests/src/main.rs#L36

**Comment by CriesofCarrots** [comment]
Changes look right to me!

File: src/background/__tests__/wallet-derivation.test.ts (hunk 2 above)
Comment (t-nelson) [reply src/background/__tests__/wallet-derivation.test.ts:45]: @fproulx-dfuse we should be able to get you some pubkeys at least.  I'll have to check whether the seed is actually made available to the app or kept hidden in the secure element

**Review by t-nelson — APPROVED** [review]
LGTM!  Thanks for addressing!

**Pull request merged**